### PR TITLE
OP_MSG support

### PR DESCRIPTION
**Pull request by tolsen**

### CODE REVIEW

File: wire.go [hunk 1]
```diff
@@ -132,3 +133,12 @@ type CommandReplyMessage struct {
 	Metadata     SimpleBSON
 	OutputDocs   []SimpleBSON
 }
+
+// OP_MSG
+// Note that checksum is not implemented
```
Comment (louisaberger) [inline wire.go]: What is this?

File: wire_message.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+	(*loc) += int(bs.Body.Size)
+
+	return bs, nil
+}
+
+func parseDocumentSequenceSection(buf []byte, loc *int) (MessageMessageSection, error) {
+	dss := &DocumentSequenceSection{}
+
+	expectedSizeRemaining := readInt32(buf[*loc:])
```
Comment (louisaberger) [inline wire_message.go]: Do we need to check here that len(buf) is >= *loc+4 so that readInt32 doesn't panic?

File: wire_message.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	docs := make([]SimpleBSON, 0)
+	docNum := 0
+	for expectedSizeRemaining > 0 {
+		doc, err := parseSimpleBSON(buf[*loc:])
+		if err != nil {
+			return dss, err
+		}
+
+		if expectedSizeRemaining < doc.Size {
+			return dss, NewStackErrorf("invalid Document Sequence section -- size of document %v (starting from 0) overruns end of section", docNum)
```
Comment (louisaberger) [inline wire_message.go]: Should we log the `expectedSizeRemaining` and `doc.Size` here too?

File: wire.go (hunk 1 above)
Comment (tolsen) [reply wire.go]: The checksum is an optional part of OP_MSG.  It would be at the end of the struct if it were implemented

**Comment by tolsen** [comment]
@louisaberger comments addressed!

**Comment by tolsen** [comment]
@erh I'm going to merge but please feel free to take a look and if you have any feedback we can address it in another PR!

**Pull request merged**